internal/testrunner/runners/system/runner.go

```diff
@@ -1603,20 +1786,93 @@ func deleteDataStreamDocs(ctx context.Context, api *elasticsearch.API, dataStrea
 	return nil
 }
 
-func filterAgents(allAgents []kibana.Agent, svcInfo servicedeployer.ServiceInfo) []kibana.Agent {
+func filterAgents(allAgents []kibana.Agent, agentInfo agentdeployer.AgentInfo, threshold time.Time, svcInfo servicedeployer.ServiceInfo, runIndependentElasticAgent bool) []kibana.Agent {
```

Comment:
Quite a lot of complexity is added here. Could we maybe base the search on the policy name, and create different policy names on each execution with independent elastic agents?